### ISSUE TITLE
The last few records in tempFiles can't flush to HDFS when stream in kafka in paused

### ISSUE THREAD

**Issue by lakeofsand** [confluentinc/kafka-connect-hdfs#79]
The key part(TopicPartitionerWriter.java):

**//when buffer.poll get the last record,but 'shouldRotate(now)' is false_,no subsequent action will occur**

`
while(!buffer.isEmpty()) {

...

SinkRecord projectedRecord = SchemaUtils.project(record, currentSchema, compatibility);
              writeRecord(projectedRecord);
              buffer.poll();
              if (shouldRotate(now)) {
                log.info("Starting commit and rotation for topic partition {} with start offsets {}"
                         + " and end offsets {}", tp, startOffsets, offsets);
                nextState();
                // Fall through and try to rotate immediately
              } else {`

**Comment by xianzhen** [comment]
We are encountering the same issue.

**Comment by Ishiihara** [comment]
That is the current design. I think the workaround here is to set `flush.interval.ms` to a reasonable value? Also, after #87 is merged, the scheduled flush can also be used as a workaround.

**Comment by skyahead** [comment]
We faced the same issue, and we tried setting flush.interval.ms but it did not help if the stream is paused. The reason is shouldRotate() is only checked inside the while(!buffer.isEmpty()) loop. But when the stream is paused, the buffer is also empty.

In our case, our stream gets data every 5 minute and we flush.size is 50K in size. So, we always have 0 to 50K records kept in temp files until new data is coming. 

Another issue we saw is that the first files per topic contain only one record, reason being lastRotate is zero and now - lastRotate >= rotateIntervalMs is always (or most likely) true.

I sent a PR to fix these issues.

Please have a look if it makes sense.

**Comment by wjptak** [comment]
We have the same issue, using the latest Confluent Kafka. Thanks for the PR!!

**Comment by cotedm** [comment]
This was resolved by #100
